src/helpers/clipboard/clipboard_figure_state.ts

```diff
@@ -84,14 +84,23 @@ export class ClipboardFigureState implements ClipboardState {
    */
   paste(target: Zone[]) {
     const sheetId = this.getters.getActiveSheetId();
-    const position = {
-      x: this.getters.getColDimensions(sheetId, target[0].left).start,
-      y: this.getters.getRowDimensions(sheetId, target[0].top).start,
-    };
-    const size = { height: this.copiedFigure.height, width: this.copiedFigure.width };
-
+    const { width, height } = this.copiedFigure;
+    const numCols = this.getters.getNumberCols(sheetId);
+    const numRows = this.getters.getNumberRows(sheetId);
+    const targetX = this.getters.getColDimensions(sheetId, target[0].left).start;
+    const targetY = this.getters.getRowDimensions(sheetId, target[0].top).start;
+    const maxX = this.getters.getColDimensions(sheetId, numCols - 1).end;
+    const maxY = this.getters.getRowDimensions(sheetId, numRows - 1).end;
+    const mainViewportRect = this.getters.getMainViewportRect();
+    const position =
+      mainViewportRect.width < width || mainViewportRect.height < height
+        ? { x: targetX, y: targetY }
```

Comment:
After thinking about this problem, I came up with few observations, 
If we try to paste a chart on the very last cell then according to the fix it should print the chart on opposite direction if there is proper viewport available. Since the available width & height of viewport is not enough for chart, current code prints it as normal case i.e. starting from same position (which results in same exact situation you found).

I propose a solution to this, but I'm not sure about how it will look, if there is no ample space in x-axis then code should reset the start point as 0th cordinate, same goes for y axis.

Reply:
When we attempt to copy in a frozen pane, we need to determine the new origin of the spreadsheet. This means that the sheet will not start from (0,0) but from some other point. We can obtain the updated (x,y) coordinates using the mainViewportRect() function if there is a frozen pane. By knowing this new coordinate, we can adjust our work accordingly and avoid creating a different behavior.